### ISSUE TITLE
Establish dead code fragments

### ISSUE THREAD

**Issue by jatkinson1000** [m2lines/convection-parameterization-in-CAM#36]
Whilst refactoring the code as part of #33 there are a few questions that have come up.

Looking at [Janni's original repo](https://github.com/yaniyuval/Neural_nework_parameterization/tree/v.1.0.3/sam_code_NN) there are a number of subdirectories under [`sam_cases/`](https://github.com/yaniyuval/Neural_nework_parameterization/tree/v.1.0.3/sam_code_NN/sam_cases), each with a different version of `nn_convection_flux.f90`, `nn_diffusion.f90`, and parameters file `prm`.

It seems that the code was designed to test a variety of configurations with/without certain variables during development through the use of switches such as `rf_uses_rh` etc. Is this behavior that needs preserving, or is there a 'final'/'best' version of the net that we are wanting to implement?

If we only want one version of the net then we can(should) remove dead code fragments to speed things up and simplify the code.

If we are wanting to implement a specific net, then which one of the instances in [`sam_cases/`](https://github.com/yaniyuval/Neural_nework_parameterization/tree/v.1.0.3/sam_code_NN/sam_cases) is it, or is it the [`nn_convection_flux.f90`](https://github.com/yaniyuval/Neural_nework_parameterization/blob/v.1.0.3/sam_code_NN/nn_convection_flux.f90) file in the `sam_code_NN/` directory?

If the latter then is this governed by the parameters listed as defaults in [`setparm.f90`](https://github.com/yaniyuval/Neural_nework_parameterization/blob/v.1.0.3/sam_code_NN/setparm.f90), as there is no `prm` parameters file in the main directory?

Finally I note that there is a file [`nn_diffusion.f90`](https://github.com/yaniyuval/Neural_nework_parameterization/blob/v.1.0.3/sam_code_NN/nn_diffusion.f90). Do we need to do anything with this? My understanding was that we were extracting `nn_convection_flux.f90`, but perhaps I have misunderstood?

**Comment by jatkinson1000** [comment]
Tagging @paogorman and @yaniyuval in the hopes that they might be able to advise.

**Comment by paogorman** [comment]
Some answers:
- We don't need to deal with nn_diffusion.f90 
- We don't need rf_uses_qp option
- We don't need rf_uses_rh in the first implementation (but we might want to add it later)

**Comment by paogorman** [comment]
I believe the correct nn_convection_flux.f90 file to use is [un_files_x8_5_layers](https://github.com/yaniyuval/Neural_nework_parameterization/tree/v.1.0.3/sam_code_NN/sam_cases/run_files_x8_5_layers)
Note this is the same as: https://github.com/yaniyuval/Neural_nework_parameterization/blob/v.1.0.3/sam_code_NN/nn_convection_flux.f90

**Comment by dorchard** [comment]
Given that we don't need the `rf_uses_qp` option, should we delete all code that pertains to that flag, i.e., that would run when this is equal to true (all the name lists set it to false at the moment)? That would leave a leaner code base to work with.

**Comment by paogorman** [comment]
Yes, that code should be deleted.

**Comment by jatkinson1000** [comment]
Thanks @paogorman 
I assume `do_yin_input` (equatorial distance) is always used?
Likewise for `Tin_feature_rf`?
If this is the case I'll refactor to remove the `if` statements and clean things up.

What about `qin_feature_rf`?

Finally @yaniyuval please could you clarify which input parameters are used for the 'final' net?
I'm assuming it's those in [setparm.f90](https://github.com/yaniyuval/Neural_nework_parameterization/blob/v.1.0.3/sam_code_NN/setparm.f90) given the absence of a `prm` file?

**Comment by paogorman** [comment]
Yes do_yin_input, Tin_feature_rf and qin_feature_rf will be always used.

The prm file is in this directory:
https://github.com/yaniyuval/Neural_nework_parameterization/tree/main/sam_code_NN/sam_cases/run_files_x8_5_layers
